packages/amplify_auth_cognito/test/amplify_auth_cognito_confirmSignup_test.dart

```diff
@@ -45,6 +45,20 @@ void main() {
     authChannel.setMockMethodCallHandler(null);
   });
 
+  test('confirmSignUp request accepts and serializes options', () async {
+    final mockKey = 'key';
+    final mockValue = 'value';
+    final mockOptions =
+        CognitoConfirmSignUpOptions(clientMetadata: {mockKey: mockValue});
+    final requst = ConfirmSignUpRequest(
+            username: 'user', confirmationCode: '123456', options: mockOptions)
+        .serializeAsMap();
+
+    expect(requst['options'], isInstanceOf<Map>());
+    expect(requst['options']['clientMetadata'], isInstanceOf<Map>());
+    expect(requst['options']['clientMetadata'][mockKey], mockValue);
```

Comment:
Nit: Would it be possible to combine `mockKey` and `mockValue` into a `mockClientMetadata` and check for that equality directly? Just think we could align it a little more with the kt/swift tests that way